bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -124,44 +140,96 @@ public AbstractRetryingOperation(
     this.retryExecutorService = retryExecutorService;
     this.originalMetadata = originalMetadata;
     this.completionFuture = new GrpcFuture<>();
+    String spanName = makeSpanName("Operation", rpc.getMethodDescriptor().getFullMethodName());
+    operationSpan = TRACER.spanBuilder(spanName).setRecordEvents(true).startSpan();
   }
 
   /** {@inheritDoc} */
   @Override
   public void onClose(Status status, Metadata trailers) {
-    synchronized (callLock) {
-      call = null;
+    try (NonThrowingCloseable s = TRACER.withSpan(operationSpan)) {
```

Comment:
doesnt this cause the span to be ended twice? once in om finalizeStats and a second time in try-with-resources?

Reply:
The API is a bit confusing.  This withSpan just sets and unsets the span in a ThreadLocal.  [Newer versions of the API](https://github.com/census-instrumentation/opencensus-java/blob/b2f48c5d608c154d75ae7616c660e053f6e5c5f7/api/src/main/java/io/opencensus/trace/SpanBuilder.java#L26) has the functionality that you expect with `tracer.spanBuilder("MyChildSpan").startScopedSpan()`.